services/metadata/backend/fake-backend/fake-backend.go

```diff
@@ -0,0 +1,192 @@
+)
+
+// Representation of a project
+type Project struct {
+	ID          string      `json:"id"`
+	Name        string      `json:"name"`
+	Description string      `json:"description"`
+	Metadata    interface{} `json:"metadata"`
+}
+
+// All projects that are being served
+var projects []Project
+
+// Full text search of a project.
+// Returns a slice of Projects where each project matches the search query.
+// Note: The query is a regex pattern and is matched against the JSON representation of the project.
+func searchProjects(query string) []Project {
+	var res []Project
+	for _, project := range projects {
+		content, _ := json.Marshal(project.Metadata)
+		match, _ := regexp.Match(query, content)
+		if match {
+			res = append(res, project)
+		}
+	}
+	return res
+}
+
+// Searches for a element with type == "http://ns.dasch.swiss/repository#Project"
+// in a json-shaped []interface{}
+func findProjectNode(list []interface{}) map[string]interface{} {
+	for _, item := range list {
+		innerMap, ok := item.(map[string]interface{})
+		if ok {
+			tp := innerMap["type"]
+			if tp == "http://ns.dasch.swiss/repository#Project" {
+				return innerMap
+			}
+		} else {
+			log.Fatal("Failed to parse node")
+		}
+	}
+	return nil
+}
+
+// Loads a project from a JSON file.
+// Expects this file to be located in ./data/*.json
+func loadProject(path string) Project {
+	// read json
+	byteValue, err := ioutil.ReadFile(path)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// unmarshal json
+	jsonMap := make(map[string]interface{})
+	err2 := json.Unmarshal(byteValue, &jsonMap)
+	if err2 != nil {
+		log.Fatal(err)
+	}
+
+	// grab actual metadata from JSON
+	projMetadata, ok := jsonMap["projectsMetadata"].([]interface{})
+	if ok {
+		projectMap := findProjectNode(projMetadata)
+		id := projectMap["shortcode"].(string)
+		name := projectMap["name"].(string)
+		description := projectMap["description"].(string)
+		return Project{
+			ID:          id,
+			Name:        name,
+			Description: description,
+			Metadata:    projMetadata,
+		}
+	} else {
+		log.Fatal("Could not find project in JSON")
+		return Project{}
+	}
+}
+
+// Load Project Data
+func loadProjectData() []Project {
+	var res []Project
+
+	paths, _ := filepath.Glob("./services/metadata/backend/fake-backend/data/*.json")
+
+	for _, path := range paths {
+		res = append(res, loadProject(path))
+	}
+
+	return res
+}
+
+// Get projects
+func getProjects(w http.ResponseWriter, r *http.Request) {
+	log.Printf("Got a request to %v", r.URL)
+
+	w.Header().Set("Content-Type", "application/json")
+	w.Header().Set("Access-Control-Expose-Headers", "X-Total-Count")
+	// TODO: are any of those needed? json-server has them
+	// w.Header().Set("Content-Type", "charset=utf-8")
+	// w.Header().Set("Cache-Control", "no-cache")
+	// w.Header().Set("Expires", "-1")
+	// w.Header().Set("Pragma", "no-cache")
+	// w.Header().Add("X-Total-Count", "10")
+	// TODO: do we need links to previous and next and first an last?
```

Comment:
FE pagination uses only first and last but if we have all options here enabled, FE can add it too.

Reply:
Ok, then I'll add them